testcases/kernel/syscalls/recvmmsg/recvmmsg01.c

```diff
@@ -0,0 +1,2389 @@
+							  callp->call_niov);
+	if (test_verbose || subtest_show())
+		tst_res(TINFO, "con_receive_iovec_boundary_checks_peeking(niov"
+			" = %d) // iov_max = %d",
+			(int) callp->call_niov, (int) iovec_max());
+	return r;
+}
+
+STATIC void call_receive_file_descriptor_init(
+		call_receive_file_descriptor_t *callp,
+		int cloexec_flag, bool some_data, ssize_t controllen_delta)
+{
+	call_base_init(&callp->call_base, call_do_con_init_socketpair,
+		       (calltest_t) call_receive_file_descriptor);
+	callp->call_cloexec_flag = cloexec_flag;
+	callp->call_some_data = some_data;
+	callp->call_controllen_delta = controllen_delta;
+}
+
+STATIC int call_receive_file_descriptor(
+		call_receive_file_descriptor_t *callp, con_t *conp, int tn)
+{
+	int r = con_receive_file_descriptor(conp, tn, callp->call_cloexec_flag,
+					    callp->call_some_data,
+					    callp->call_controllen_delta);
+	if (test_verbose || subtest_show())
+		tst_res(TINFO, "con_receive_file_descriptor("
+			"cloexec_flag = %s, some_data = %d, "
+			"controllen_delta = %ld)",
+			(callp->call_cloexec_flag & MSG_CMSG_CLOEXEC) ?
+				"MSG_CMSG_CLOEXEC" : "0",
+			callp->call_some_data,
+			(long) callp->call_controllen_delta);
+	return r;
+}
+
+STATIC void call_msgtoolong_do_con_init_from_type(
+	call_message_too_long_to_fit_might_discard_bytes_t *callp,
+	con_t *conp)
+{
+	con_init_with_type(conp, callp->call_type);
+}
+
+STATIC void call_message_too_long_to_fit_might_discard_bytes_init(
+		call_message_too_long_to_fit_might_discard_bytes_t *callp,
+		int type)
+{
+	call_base_init(&callp->call_base,
+		       (callconinit_t) call_msgtoolong_do_con_init_from_type,
+		       (calltest_t)
+		       call_message_too_long_to_fit_might_discard_bytes);
+	callp->call_type = type;
+}
+
+STATIC int call_message_too_long_to_fit_might_discard_bytes(
+		call_message_too_long_to_fit_might_discard_bytes_t *callp,
+		con_t *conp, int tn)
+{
+	int r = con_message_too_long_to_fit_might_discard_bytes(conp, tn,
+							callp->call_type);
+	if (test_verbose || subtest_show())
+		tst_res(TINFO,
+			"con_message_too_long_to_fit_might_discard_bytes"
+			"(type = %s)", sock_type_to_str(callp->call_type));
+	return r;
+}
+
+STATIC void call_empty_datagram_received_as_empty_datagram_init(
+		call_empty_datagram_received_as_empty_datagram_t *callp)
+{
+	call_base_init(&callp->call_base, call_do_con_init_dgram,
+		       (calltest_t)
+		       call_empty_datagram_received_as_empty_datagram);
+}
+
+STATIC int call_empty_datagram_received_as_empty_datagram(
+		call_empty_datagram_received_as_empty_datagram_t *callp,
+		con_t *conp, int tn)
+{
+	unused(callp);
+	int r = con_empty_datagram_received_as_empty_datagram(conp, tn);
+	if (test_verbose || subtest_show())
+		tst_res(TINFO,
+			"con_empty_datagram_received_as_empty_datagram()");
+	return r;
+}
+
+#ifdef RECVMMSG_USE_SCTP /*{*/
+
+STATIC void call_receive_waits_for_message_init(
+		call_receive_waits_for_message_t 
```

Comment:
You are using this function only once, there is no point in having this function at all, it only obfuscates the actual code.